test/custom_resource_crud/custom_resource_crud_test.go

```diff
@@ -73,10 +74,205 @@ var _ = Describe("crcrud test", func() {
+				if err != nil {
+					testenvInstance.Log.Error(err, "Unable to fetch Indexer Cluster deployment")
+				}
+				idxc.Spec.Resources.Limits = corev1.ResourceList{
+					"cpu": resource.MustParse(newCPULimits),
+				}
+				err = deployment.UpdateCR(idxc)
+				Expect(err).To(Succeed(), "Unable to deploy Indexer Cluster with updated CR")
+			}
+
+			// Verify Indexer Cluster is updating
+			idxcName := deployment.GetName() + "-" + "site1"
+			testenv.VerifyIndexerClusterPhase(deployment, testenvInstance, splcommon.PhaseUpdating, idxcName)
+
+			// Verify Indexers go to ready state
+			testenv.IndexersReady(deployment, testenvInstance, siteCount)
+
+			// Verify CPU limits after updating the CR
+			for i := 1; i <= siteCount; i++ {
+				podName := fmt.Sprintf(testenv.MultiSiteIndexerPod, deployment.GetName(), i, 0)
+				testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), podName, newCPULimits)
+			}
+
+			// Verify CPU limits on Search Heads before updating the CR
+			SearchHeadPodName := fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), 0)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), SearchHeadPodName, defaultCPULimits)
+			SearchHeadPodName = fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), 1)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), SearchHeadPodName, defaultCPULimits)
+			SearchHeadPodName = fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), 2)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), SearchHeadPodName, defaultCPULimits)
+
+			// Change CPU limits to trigger CR update
+			shc := &enterprisev1.SearchHeadCluster{}
+			instanceName := fmt.Sprintf("%s-shc", deployment.GetName())
+			err = deployment.GetInstance(instanceName, shc)
+			if err != nil {
+				testenvInstance.Log.Error(err, "Unable to fetch Search Head Cluster deployment")
+			}
+
+			shc.Spec.Resources.Limits = corev1.ResourceList{
+				"cpu": resource.MustParse(newCPULimits),
+			}
+			err = deployment.UpdateCR(shc)
+			Expect(err).To(Succeed(), "Unable to deploy Search Head Cluster with updated CR")
+
+			// Verify Search Head Cluster is updating
+			testenv.VerifySearchHeadClusterPhase(deployment, testenvInstance, splcommon.PhaseUpdating)
+
+			// Verify Search Head go to ready state
+			testenv.SearchHeadClusterReady(deployment, testenvInstance)
+
+			// Verify CPU limits after updating the CR
+			SearchHeadPodName = fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), 0)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), SearchHeadPodName, newCPULimits)
+			SearchHeadPodName = fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), 1)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), SearchHeadPodName, newCPULimits)
+			SearchHeadPodName = fmt.Sprintf(testenv.SearchHeadPod, deployment.GetName(), 2)
+			testenv.VerifyCPULimits(deployment, testenvInstance.GetName(), SearchHeadPodName, newCPULimits)
```

Comment:
Do not need to repeat this here as SHC update is already covered in C3 deployment